src/backend/vulkan/src/lib.rs

```diff
@@ -437,7 +433,6 @@ impl hal::Instance<Backend> for Instance {
                 })
             });
 
-        #[cfg(debug_assertions)]
         let debug_messenger = {
             // make sure VK_EXT_debug_utils is available
             if instance_extensions.iter().any(|props| unsafe {
```

Comment:
there is a change missing here. We don't want to request the `VK_EXT_debug_utils` or `DebugReport::name()` if running in release. So these conditions should look like:
```rust
if cfg!(debug_assertions) && instance_extensions.iter().any(...) {
  ...
} else if cfg!(debug_assertions) && instance_extensions.iter().any(..) {
  ...
} else {
  None
}
```

Reply:
If we don't request debug utils, then [insert_debug_marker will be a no-op.](https://github.com/gfx-rs/gfx/blob/e99e71d0022f72fe4f558ce9f528e258131af2a5/src/backend/vulkan/src/command.rs#L1121)

Reply:
Oh snap, you are right. In this case, we should at least gate the `DebugReport::name()` by `debug_assertions`?